asapdiscovery-workflows/asapdiscovery/workflows/docking_workflows/ligand_transfer_docking.py

```diff
@@ -0,0 +1,314 @@
+        targets,
+        use_dask=inputs.use_dask,
+        dask_client=dask_client,
+        dask_failure_mode=inputs.dask_failure_mode,
+        cache_dir=inputs.cache_dir,
+        use_only_cache=inputs.use_only_cache,
+    )
+    del targets
+
+    n_prepped_complexes = len(prepped_complexes)
+    logger.info(f"Prepped {n_prepped_complexes} complexes")
+    for pc in prepped_complexes:
+        logger.info(f"Complex is {pc.target.target_name}")
+
+    if inputs.save_to_cache and inputs.cache_dir is not None:
+        logger.info(f"Writing prepped complexes to global cache {inputs.cache_dir}")
+        prepper.cache(prepped_complexes, inputs.cache_dir)
+
+    # Here the only thing that makes sense is the self docking selector
+    selector = StructureSelector.SELF_DOCKING.selector_cls()
+    ligands = [pc.ligand for pc in prepped_complexes]
+    pairs = selector.select(ligands, prepped_complexes)
+
+    n_pairs = len(pairs)
+    logger.info(f"Selected {n_pairs} pairs for docking, from {len(ligands)} ligands and {n_prepped_complexes} compexes")
+    del prepped_complexes
+
+    # dock pairs
+    logger.info("Running docking on selected pairs")
+    docker = POSITDocker(
+        relax=inputs.relax,
+        posit_method=inputs.posit_method,
+        use_omega=inputs.use_omega,
+        omega_dense=inputs.omega_dense,
+        num_poses=inputs.num_poses,
+        allow_low_posit_prob=inputs.allow_low_posit_prob,
+        low_posit_prob_thresh=inputs.low_posit_prob_thresh,
+        allow_final_clash=inputs.allow_final_clash,
+        allow_retries=inputs.allow_retries,
+    )
+    results = docker.dock(
+        pairs,
+        output_dir=output_dir / "docking_results",
+        use_dask=inputs.use_dask,
+        dask_client=dask_client,
+        dask_failure_mode=inputs.dask_failure_mode,
+    )
+
+    n_results = len(results)
+    logger.info(f"Docked {n_results} pairs successfully")
+    if n_results == 0:
+        raise ValueError("No docking results generated, exiting")
+    del pairs
+
+    # add chemgauss4 scorer
+    scorers = [ChemGauss4Scorer()]
+
+    # score results
+    logger.info("Scoring docking results")
+    scorer = MetaScorer(scorers=scorers)
+
+    if inputs.write_final_sdf:
+        logger.info("Writing final docked poses to SDF file")
+        write_ligands_to_multi_sdf(
+            output_dir / "docking_results.sdf", [r.posed_ligand for r in results]
+        )
+    # NOTE: The return_df option doesn't work, the DataFrame.pivot() function fails. Fix is pending.
+    scores_list = scorer.score(
+        results,
+        use_dask=inputs.use_dask,
+        dask_client=dask_client,
+        return_df=False,
+        dask_failure_mode=inputs.dask_failure_mode,
+    )
+    del results
+    import csv
+    with open(data_intermediates / "docking_scores_raw.csv", 'w', newline='') as myfile:
+        wr = csv.writer(myfile, quoting=csv.QUOTE_ALL)
+        for item in scores_list:
+            wr.writerow([item]) 
```

Comment:
@mariacm12 could you add info on the error you got? is it the target-related stuff?

Reply:
It's because the dataframe has duplicate entries I think. Although I couldn't see the duplicates myself when I generated the csv file the alternative way (from DataFrame.pivot() docs: _Raises ValueError when there are any index, columns combinations with multiple values_)